_proposals/pup-0001.rst

```diff
@@ -0,0 +1,328 @@
+
+
+Is This Replacing Story Planning in Redmine?
+============================================
+
+No. This is for changes to process and should not be used to discuss the feature set or bugs of
+Pulp. Feature and bug discussion would continue to occur as-is (on pulp.plan.io). Two recent-ish
+examples of things which could have gone through this process include "Napoleon docstrings or not?",
+"To Merge forward or not?".
+
+This is different than the communities we are modeling our process after. They use it for feature
+planning also, but I propose that we don't do that currently.
+
+
+Examples
+========
+
+* This proposal
+* All the PEPs in both [web](https://www.python.org/dev/peps/) and
+  [raw](https://github.com/python/peps) form.
+
+
+Drawbacks
+=========
+
+* The history is in two places: the commit history and separately the mailing list archive.
+* Initial creation and revisions require merging by a core dev [#f1]_ versus having it live as a PR
+  at all times.
+* Mailing lists can be hard to follow.
+
+
+Alternatives
+============
+
+We could store the PEPs in Redmine. Why aren't we using Redmine?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upsides:
+
+* It would keep every comment and the proposal itself on one webpage.
+* It would be kept along with bugs and stories
+
+Downsides:
+
+* Redmine isn't great for long, structured documents. For instance linking between sections is hard.
+* Redmine's long conversations become difficult to read
+* Comments don't support threading well.
+* History of Redmine edits is not as clear as a git commit history.
+* Not as beautiful as a web-viewable archive of decisions
+
+
+What about leaving it as a pull request for longer?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upsides:
+
+* Anyone can create and maintain a proposal with complete self-service until it needs to be merged.
+
+Downsides:
+
+* This is very developer focused. Everyone knows how use emails and a web browser. Not everyone
+  knows how to use Github review.
+* The proposal never being merged to a Pulp repo may cause us to loose access to previous proposals.
+* Maintaining a numbered record of Proposals is more difficult.
+
+
+What about using Github for discussion?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Upsides:
+
+* It keeps comments in one place, with the proposal document
+* You can leave comments line-by-line
+* It supports threading pretty well
+* You could use github voting.
+
+Downside:
+
+* This is very developer focused. Everyone knows how use emails and a web browser. Not everyone
+  knows how to use Github review.
+* It breaks up the document which makes it practically less readable
+* It can be hard to see comments over time with new versions collapsing the comments when they
+  become "outdated".
+
+
+Unresolved Questions
+====================
+
+What are we calling it? (PUP, PCP, <your idea here>)
+Is 10 calendar days the right decision making time default.
```

Comment:
I think I'd prefer two weeks, but agree that making this much longer would be counter productive.